src/components/common/button/index.js

```diff
@@ -0,0 +1,40 @@
+/** @jsx jsx */
+import { jsx } from '@emotion/core'
+import React from 'react'
+import * as ReactRouter from 'react-router-dom'
+
+import Flex from 'components/common/flex'
+
+import style from './index.style.js'
+
+class Button extends React.Component {
+  render() {
+    return (
+      <Flex
+        css={style.base}
+        {...this.props}
+        onClick={this.handleClick}
+        type="button"
+      />
+    )
+  }
+
+  handleClick = event => {
+    const { action, history, onClick, to } = this.props
+    console.log(this.props)
+    if (onClick) {
+      onClick(event)
+    }
+    if (action === 'goBack') {
+      history.goBack()
+    }
+    if (action === 'externalLink') {
+      window.location = to
+    }
+    if (action === 'internalLink') {
+      history.push(to)
+    }
```

Comment:
Just see which props are set and take action depending on that.

You could have a `to` and `url` props that we would use to navigate internally and externally

Reply:
refactored to `  handleClick = () => {
    const { history, onClick, to, url } = this.props
    if (onClick) {
      onClick()
    }
    if (url) {
      window.location = url
    }
    if (to) {
      history.push(to)
    }
  }`